draft-ietf-quic-transport.md

```diff
@@ -2618,6 +2633,39 @@ advertise a smaller maximum ID.  A sender may receive MAX_STREAM_ID frames out
 of order; a sender MUST therefore ignore any MAX_STREAM_ID that does not
 increase the maximum.
 
+
+## Related Streams
+
+Using the Related Stream ID in the STREAM frame ({{frame-stream}}) allows for
+the creation of a pair of streams.  One endpoint creates a stream without
+assigning a related stream and the peer opens another stream that includes the
+stream ID of the first in the Related Stream ID of the first stream.  This
+enables straightforward use of QUIC for request-response protocols or protocols
+that rely on bidirectional channels.
+
+This document doesn't proscribe the use of the Related Stream ID field, leaving
+decisions about the semantics of related streams to protocols that use QUIC.
+Protocols that use QUIC MUST either define the semantics of related streams or
+prohibit the use of the Related Stream ID field.  Protocols MAY define other
+uses for the Related Stream ID field, but are advised that complex relationships
+between streams could be poorly supported in generic QUIC implementations.  The
+INVALID_RELATED_STREAM error code can be used by protocols to signal receipt of
+invalid Related Stream ID field values, unless the protocol defines a more
+specific error code for the circumstances.
+
+The Related Stream ID for a stream only needs to be sent once for each (allowing
+for retransmissions due to loss).  Sending the field on all stream frames with
+an offset of 0 ensures that the value is received without duplicating the field
+unnecessarily.  If the value of the Related Stream ID changes for a stream, the
+endpoint that receives the conflicting value MUST treat this as a connection
+error of type INVALID_RELATED_STREAM.
```

Comment:
First part of paragraph could be shortened a bit to just state the related stream id is only sent in the first frame with offset zero (allowing for retransmission).

Second part of the paragraph is unclear - what does it mean to receive different ID's? It can only happen during retransmission because offset 0 can only be sent once. If due to retransmission, the protocol logic should only see one version and discard duplicates without interpretation (general protocol logic, though this might be not be explicit).

The second part could also be interpreted as if the original referenced stream can only be associated with a single stream and using multiple would be an error. But I do not believe this is the intention, and if it is, I don't think it is reasonable to place any restrictions here. But - it might be good to add a note stating the a field may be referenced 0, 1, or more times depending on application semantics.